### ISSUE TITLE
Edmond Brain Dump

### ISSUE THREAD

**Issue by Edmond5858** [cryptovoxels/issues#37]
Hey guys. Spidy invited me to capture some of my random ideas here so I will just edit and add to this doc over time.

**TOP TOP TOP PRIORITY IMO**

**• Undo function** 

**• Fix out-of-bounds build-ability.** Currently a lot of the building i have to do is from perspective of neighboring parcel. It's possible to build with current tool but when you want to change tools you have to fly inside of parcel to select new tool which is annoying if you have a wall in front of you. Also a problem on the roof. 

**USABILITY**

• Drag Vertical to place multiple blocks - i believe you might be working on this already

• Ability to ascend/descend while flying without looking in that direction. I thought space and command made sense for this but others seemed to think Q/E was the standard. I like the idea of using thumb though so you can also make all other movements simultaneously.

• Allow us to swap blocks so we don't have to remove and then place.

• Need a coloring tool. I think this feature might be the same as the swapping blocks. But need a very quick way to colorize with a paint tool.

• Need HUD so you know what tool you have currently selected.

• Need much quicker way to swap tools without needing to go into a secondary menu. Currently there is just too much clicking around in menus while building. 

* 1 button to toggle between inverted blocks while buliding… or double click on swatch. Inverted button is so far from blocks right now. Annoying to use and you need laser precision to check that box.

* Add interactivity to in-world map. Need to see where users are at. Need to be able to search and warp anywhere without going back to CV website

**AESTHETICS**

• Add plain glass texture with no grid lines

* Newly minted land should default with nothing on it. Otherwise it’s annoying to have your property blocked by HODLer scaffolding

**BUGS/FIXES**

**• Fix dropbox linking.** In script replace "www.dropbox.com" with "dl.dropboxusercontent.com" for vox and image linking. Source: https://ryanmo.co/2013/11/03/dropboxsharedlinks/

* Currently the drag to place multiple blocks will overwrite other preexisting blocks. This is frustrating without undo feature.

* Looking around on my iPhone is hyper sensitive. I’m a gamer and am already used to first person shooters where i use fairly high sensitivity. Even with that experience, this is unusable for me. I think sensitivity needs to go way down.

**FUTURE MISC**

* Add consequences of some sort for someone who buys and doesn’t build on the land. Something that would make the parcel less valuable for them (maybe parcel COLR goes into negative over time). Need to encourage building. Either that or instead of punishing non-builders, incentivize builders with something. Lots of free games offer rewards for logging in several days in a row. Maybe building 5 days in a row would reward with you COLR or unlocks some other feature.

* Streaming localized music/video without needing to press play

* Add a tile customizer where you can create pixel art and turn it into a custom box texture without leaving CV

* Stake COLR to raise build height (would have to limit this so it’s not skyscrapers everywhere. But sometimes if you own two adjacent buildings they are like 2-3 vox off on the roofline. I think if you owned both it would be nice to have option to even that up.  

* Or another option to stake COLR to dig deeper underground. I think underground should be very deep but expensive so that everyone wouldn’t have ridiculous basements.  

* Piggybacking off the dig idea, i think it would be great to be able to connect parcels underground beneath roadways. 

* If you own adjacent properties that are perfectly aligned with a road separating them, I think you should be able to upgrade and combine that into one parcel to basically take over the roadway and build over it. All of the blocks are so small now. We need the option for much greater scale and a more diverse street grid.

* If no option to combine properties then i think we need a few warehouse industrial districts with low build height but much much larger width/depth scale

* Need park districts with very very low build height. Like 4 voxes. Not enough to put a structure up but enough to put small things like markets and benches and landscaping,  

* We need to have a plan for adult content. I for one wouldn’t want my wife or kid coming into CV and seeing that stuff all over. Needs to be quarantined to a district and somehow have age limitations on your avatar that lock you out  

• Add permissions to parcel contributors. So for instance you can allow people to update images and vox models but not to change the structure at all.

**GAMING** - Ben asked for some brainstorming on this

* NFT lock/keys and combination locks to hide/reveal areas

* Fair games, bowling, knocking over milk bottles, etc in VR

* Scavenger hunts for rewards (my 4 Render Extension property was built with this in mind. You can see that an object is missing and seems like you’re supposed to find it and place it in the scene). The idea was I could have it randomly in one of my other properties and if you collect it and place then you get something.

* Flying or race challenges through rings

* Paint by number

* Something similar to Splatoon that colorizes areas with weapons. 

* I always thought minesweeper would be neat in this world lol

* Puzzle games like Pipedream or https://assetstore.unity.com/packages/templates/packs/unblock-the-ball-sliding-puzzle-88562

* VR rubix cube

* Tetris huge on side or top of building - I also think many old atari games would be cool to play huge on the side of a building as 3d vox. Missile command, pacman, galaga, breakout, etc. 

* Escape room type scenarios. Ex: sudoku on a wall unlocks or rewards you with something once solved.

* 3d Angry Birds type game would fit well… or something like the old game Worms

* destructive mode similar to the old game Rampage where you are huge monsters and basically destroy the city
